### PR TITLE
teaser component

### PR DESCRIPTION
**Pull request by Rajamouli-hcl**
Added teasercomponent

### CODE REVIEW

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/teaser/_cq_dialog/.content.xml [hunk 1]
```diff
@@ -0,0 +1,352 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/teaser/_cq_dialog/.content.xml:1]: @Rajamouli-hcl : what are the changes in dialog as you have copied from core here??

File: ui.content/src/main/content/jcr_root/conf/hclecomm/settings/wcm/policies/.content.xml [hunk 2]
```diff
@@ -265,6 +265,17 @@
...
                     </policy_1588584656785>
                 </container>
+				<teaser jcr:primaryType="nt:unstructured">
+                    <policy_1588926494157
+                        jcr:lastModified="{Date}2020-05-08T15:27:31.381+05:30"
+                        jcr:lastModifiedBy="admin"
+                        jcr:primaryType="nt:unstructured"
+                        jcr:title="Teaser policy"
+                        sling:resourceType="wcm/core/components/policy/policy"
+                        titleLinkHidden="true">
```
Comment (rishabh-behl) [inline ui.content/src/main/content/jcr_root/conf/hclecomm/settings/wcm/policies/.content.xml]: @Rajamouli-hcl : why do we need titleLinkHidden in policy?

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/teaser/.content.xml [hunk 3]
```diff
@@ -1,8 +1,7 @@
...
-<jcr:root xmlns:sling="http://sling.apache.org/jcr/sling/1.0" xmlns:cq="http://www.day.com/jcr/cq/1.0"
-    xmlns:jcr="http://www.jcp.org/jcr/1.0"
+<jcr:root xmlns:sling="http://sling.apache.org/jcr/sling/1.0" xmlns:cq="http://www.day.com/jcr/cq/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0"
     jcr:primaryType="cq:Component"
     jcr:title="Teaser"
     sling:resourceSuperType="core/wcm/components/teaser/v1/teaser"
     componentGroup="hclecomm.Content"
-    imageDelegate="hclecomm/components/content/image"/>
\ No newline at end of file
+    imageDelegate="hclecomm/components/content/image"/>
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/teaser/.content.xml:7]: @Rajamouli-hcl : does anything break if u remove imageDelegate property ?

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/teaser/clientlibs/less/teaser.less [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+                        &:hover{
+                            color:#fff;
+                            text-decoration:none;
+                        }
+                    }
+        }
+        
+        .cmp-teaser__description{
+            font-size:32px;
+
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/teaser/clientlibs/less/teaser.less]: @Rajamouli-hcl  @ashishgoyalweb : can see lot of whitespace .. please refactor

**Review by rishabh-behl — COMMENTED** [review]
@Rajamouli-hcl  @ashishgoyalweb  Please incorporate comments

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/teaser/_cq_dialog/.content.xml (hunk 1 above)
Comment (Rajamouli-hcl) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/teaser/_cq_dialog/.content.xml:1]: added node for pretitle which is not prasent core component

File: ui.content/src/main/content/jcr_root/conf/hclecomm/settings/wcm/policies/.content.xml (hunk 2 above)
Comment (Rajamouli-hcl) [reply ui.content/src/main/content/jcr_root/conf/hclecomm/settings/wcm/policies/.content.xml]: when we select this the teaser title is not link to the linked page

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/teaser/.content.xml (hunk 3 above)
Comment (Rajamouli-hcl) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/teaser/.content.xml:7]: component is not working

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/teaser/clientlibs/less/teaser.less (hunk 4 above)
Comment (Rajamouli-hcl) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/teaser/clientlibs/less/teaser.less]: changes updated

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/teaser/teaser.html [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+     data-sly-use.imageTemplate="image.html"
+     data-sly-use.titleTemplate="title.html"
+     data-sly-use.descriptionTemplate="description.html"
+     data-sly-use.actionsTemplate="actions.html"
+     data-sly-test.hasContent="${teaser.imageResource || teaser.title || teaser.description || teaser.actions.size > 0}"
+     class="cmp-teaser${!wcmmode.disabled && teaser.imageResource ? ' cq-dd-image' : ''}">
+    <sly data-sly-call="${imageTemplate.image @ teaser=teaser}"></sly>
+
+    <div class="cmp-teaser__content">
+        <sly>${properties.preTitle}</sly>
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/teaser/teaser.html]: @Rajamouli-hcl : why is the pre-title inluded without div tag .

<div class="cmp-teaser__pretitle">Pretitle</div>

Pl refere to component dev.

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/teaser/_cq_dialog/.content.xml (hunk 1 above)
Comment (rishabh-behl) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/teaser/_cq_dialog/.content.xml:1]: @Rajamouli-hcl : why do u want to copy yje whole dilaog when u can overy. and add single node. Ref below:
image

File: ui.content/src/main/content/jcr_root/conf/hclecomm/settings/wcm/policies/.content.xml (hunk 2 above)
Comment (rishabh-behl) [reply ui.content/src/main/content/jcr_root/conf/hclecomm/settings/wcm/policies/.content.xml]: @Rajamouli-hcl : Title wont be linked until a URL is specified in dialog . Is it not the case ? I removed this policy and still title is not a link unless URL added

image

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/teaser/teaser.html (hunk 5 above)
Comment (Rajamouli-hcl) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/teaser/teaser.html]: i will replace sly tag with div.

File: ui.content/src/main/content/jcr_root/conf/hclecomm/settings/wcm/policies/.content.xml (hunk 2 above)
Comment (Rajamouli-hcl) [reply ui.content/src/main/content/jcr_root/conf/hclecomm/settings/wcm/policies/.content.xml]: yes correct rishabh, i thought this story  even if the url is specified in dialog that won't be linked with title for this i added policy,if it's not the case please confirm i will remove policy

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/teaser/_cq_dialog/.content.xml (hunk 1 above)
Comment (Rajamouli-hcl) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/teaser/_cq_dialog/.content.xml:1]: i can't do overlay for core components that's way i copied and used the whole dialog
 as per your reference  i checked its working i will do these changes

File: ui.content/src/main/content/jcr_root/conf/hclecomm/settings/wcm/policies/.content.xml (hunk 2 above)
Comment (rishabh-behl) [reply ui.content/src/main/content/jcr_root/conf/hclecomm/settings/wcm/policies/.content.xml]: @Rajamouli-hcl : Pl remove this policy

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/teaser/teaser.html (hunk 5 above)
Comment (rishabh-behl) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/teaser/teaser.html]: @Rajamouli-hcl  : cant still see the update ..

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/teaser/_cq_dialog/.content.xml (hunk 1 above)
Comment (Rajamouli-hcl) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/teaser/_cq_dialog/.content.xml:1]: changes has been commited as per the review comments

File: ui.content/src/main/content/jcr_root/conf/hclecomm/settings/wcm/policies/.content.xml (hunk 2 above)
Comment (Rajamouli-hcl) [reply ui.content/src/main/content/jcr_root/conf/hclecomm/settings/wcm/policies/.content.xml]: policy has been removed as per the review comments

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/teaser/teaser.html (hunk 5 above)
Comment (Rajamouli-hcl) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/teaser/teaser.html]: sly tag is replaced with div tag as per the comments

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/teaser/teaser.html [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+     data-sly-use.imageTemplate="image.html"
+     data-sly-use.titleTemplate="title.html"
+     data-sly-use.descriptionTemplate="description.html"
+     data-sly-use.actionsTemplate="actions.html"
+     data-sly-test.hasContent="${teaser.imageResource || teaser.title || teaser.description || teaser.actions.size > 0}"
+     class="cmp-teaser${!wcmmode.disabled && teaser.imageResource ? ' cq-dd-image' : ''}">
+    <sly data-sly-call="${imageTemplate.image @ teaser=teaser}"></sly>
+
+    <div class="cmp-teaser__content">
+        <div>${properties.preTitle}</div>
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/teaser/teaser.html]: @Rajamouli-hcl : The div needs to be like below , I provided link in prev comment too : <div class="cmp-teaser__pretitle">Pretitle</div>

https://www.aemcomponents.dev/content/core-components-examples/library/page-authoring/teaser.html

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/clientlibs/clientlib-base/.content.xml [hunk 7]
```diff
@@ -3,4 +3,4 @@
     jcr:primaryType="cq:ClientLibraryFolder"
     allowProxy="{Boolean}true"
     categories="[hclecomm.base]"
-    embed="[core.wcm.components.accordion.v1,core.wcm.components.tabs.v1,core.wcm.components.carousel.v1,core.wcm.components.image.v2,core.wcm.components.breadcrumb.v2,core.wcm.components.search.v1,core.wcm.components.form.text.v2,hclecomm.grid,hclecomm.title,hclecomm.text,hclecomm.button,hclecomm.accordion,hclecomm.image,hclecomm.carousel]"/>
+	embed="[core.wcm.components.accordion.v1,core.wcm.components.tabs.v1,core.wcm.components.carousel.v1,core.wcm.components.image.v2,core.wcm.components.breadcrumb.v2,core.wcm.components.search.v1,core.wcm.components.form.text.v2,hclecomm.grid,hclecomm.title,hclecomm.text,hclecomm.button,hclecomm.accordion,hclecomm.image,hclecomm.teaser,hclecomm.carousel]"/>
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/clientlibs/clientlib-base/.content.xml]: @Rajamouli-hcl : is there a spacae added here ? pl remove

**Review by rishabh-behl — COMMENTED** [review]
@Rajamouli-hcl : Pl incorporate comments

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/teaser/teaser.html (hunk 6 above)
Comment (Rajamouli-hcl) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/components/content/teaser/teaser.html]: div structure changed as per the review comments

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/clientlibs/clientlib-base/.content.xml (hunk 7 above)
Comment (Rajamouli-hcl) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/clientlibs/clientlib-base/.content.xml]: spaces removed please check

**Review by rishabh-behl — APPROVED** [review]
LGTM

**Pull request merged**